File: backend/src/main/java/com/coing/util/BasicResponse.java
package com.coing.util;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import com.coing.global.exception.BusinessException;
import com.coing.global.exception.doc.ErrorCode;

public record BasicResponse(
        HttpStatus status,
        String message,
        String detail
) {
    public static BasicResponse of(BusinessException exception) {
        return new BasicResponse(
                exception.getStatus(),
                exception.getMessageCode(),
                exception.getDetail()
        );
    }

    public static BasicResponse of(String message) {
        return new BasicResponse(
                HttpStatus.OK,
                message,
                ""
        );
    }

    public static ResponseEntity<BasicResponse> to(BusinessException exception) {
        return ResponseEntity
                .status(exception.getStatus())
                .body(BasicResponse.of(exception));
    }

    public static BasicResponse from(ErrorCode errorCode, MessageUtil messageUtil) {
        return new BasicResponse(errorCode.getStatus(),
            messageUtil.resolveMessage(errorCode.getMessageKey()),
            "");
    }
}

File: backend/src/main/java/com/coing/util/BasicResponse.kt
package com.coing.util

import com.coing.global.exception.BusinessException
import com.coing.global.exception.doc.ErrorCode
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity

data class BasicResponse(
    val status: HttpStatus,
    val message: String,
    val detail: String
) {
    companion object {
        fun of(exception: BusinessException): BasicResponse {
            return BasicResponse(
                exception.status,
                exception.messageCode,
                exception.detail
            )
        }

        fun of(message: String): BasicResponse {
            return BasicResponse(
                HttpStatus.OK,
                message,
                ""
            )
        }

        fun to(exception: BusinessException): ResponseEntity<BasicResponse> {
            return ResponseEntity
                .status(exception.status)
                .body(of(exception))
        }

        fun from(errorCode: ErrorCode, messageUtil: MessageUtil): BasicResponse {
            return BasicResponse(
                errorCode.status,
                messageUtil.resolveMessage(errorCode.messageKey),
                ""
            )
        }
    }
}
